packages/table/src/common/utils.ts

```diff
@@ -234,11 +234,13 @@ export const Utils = {
         } else if (objA == null || objB == null) {
             return false;
         } else if (Array.isArray(objA) || Array.isArray(objB)) {
-            return Utils.arraysEqual(objA, objB);
+            return Utils.arraysEqual(objA, objB, Utils.deepCompareKeys);
         } else if (typeof objA === "string" || typeof objB === "string" ) {
             return objA === objB;
         } else if (typeof objA === "number" || typeof objB === "number" ) {
             return objA === objB;
+        } else if (typeof objA === "boolean" || typeof objB === "boolean" ) {
+            return objA === objB;
```

Comment:
ok so we now have three identical statements here, if either object is a primitive type. refactor perhaps?